### PR TITLE
supporting "percentage fee" additional items

### LINKED ISSUE #1283: Ability to have booking fee a percentage of ticket price
**Is your feature request related to a problem? Please describe.**
Currently we can add a booking fee as a fixed price Additional Option and make it Mandatory - One for each ticket purchased. It's common for us to have different priced tickets and it would be useful to be able to have the booking fee a percentage of the ticket price.

**Describe the solution you'd like**
What would be perfect would be to have a radio button on the additional options dialogue to select between "fixed price" and "percentage of ticket price"

**Describe alternatives you've considered**
Another option would be to have a separate admin area for booking fees and have options for fixed price or percentage, If it's tricky to implement within the current additional options framework.

**Additional context**
Let's say we offer a high priced VIP week long ticket for say £200 - We want to cover the credit card processing and platform fee on this large price so set the booking fee to something high like £10. 

Now imagine we offer a very cheap ticket for children on just one day at £10. now that £10 booking fee is very high.

This is a feature available on all the commercial ticketing platforms I've used so feel it's a worthwhile addition.

Thanks for reading! x

### PR DESCRIPTION
**Pull request by cbellone**
fix #1283 

Sponsored by [Offbeat Tickets](https://offbeattickets.com)

### CODE REVIEW

File: frontend/admin/src/service/event.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {EventWithOrganization} from "../model/event.ts";
+
+export class EventService {
+    static load(publicIdentifier: string): Promise<EventWithOrganization> {
+        return fetch(`/admin/api/events/${publicIdentifier}`)
```
Comment (syjer) [inline frontend/admin/src/service/event.ts:5]: use fetchJson helper?

**Review by syjer — APPROVED** [review]

File: .github/workflows/build-on-push.yml [hunk 2]
```diff
@@ -9,8 +9,8 @@ jobs:
             matrix:
                 postgresql: ['10','16']
```
Comment (syjer) [inline .github/workflows/build-on-push.yml:10]: also add the '15' target?

**Comment by crutchcorn** [comment]
Hey all!

I'm the maintainer of TanStack Form (which seems to've been introduced in this PR).

As far as I can tell, y'all are the open-source project with the most stars currently using Form.

Just wanted to drop a line, say "thanks" for picking our tool, and see how the experience was, and if there's anything in particular you'd want us to sure up prior to our 1.x release. 😊

In particular, I know our Lit docs are lacking - we have [a ticket to update that](https://github.com/TanStack/form/issues/747).

Also, I noticed that y'all are using Angular in your repo as well. Is there any chance y'all might be similarly interested in using TanStack Form's Angular adapter? We even had the official Angular core team help me code review that adapter recently:

https://tanstack.com/form/latest/docs/framework/angular/quick-start

Best,

\- Corbin

**Comment by syjer** [comment]
hi @crutchcorn , first of all, thank you for creating this nice library! We are currently doing a modernization of our very old angularjs "admin" frontend, this PR is the first of this big work. As a first impression it seems to work as expected, as we are more familiar with the ReactiveForm which is provided out of the box in angular. 

The only issue that we have noticed, but we are not really sure if it's problematic is a warning which is printed in the console "Element XXXXXXX scheduled an update (generally because a property was set) after an update completed, causing a new update to be scheduled. This is inefficient and should be avoided unless the next update can only be scheduled as a side effect of the previous update. See https://lit.dev/msg/change-in-update for more information.".

You can see the exact same warning when trying https://github.com/TanStack/form/tree/main/examples/lit/simple , on the first load. Maybe it's a known issue?

**Comment by crutchcorn** [comment]
@syjer FWIW we get similar updates in React as well, which is a bit of a headache. The challenge is that the frameworks aren't in control of reactivity with our library; our own internal signals implementation is. As a result, there are occasional warnings about things like this from the framework as it seems to get confused about our control of reactivity. AFAIK, that warning should be easily ignored.

**Pull request merged**